infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationManager.java

```diff
@@ -217,6 +205,7 @@ public void processStandbyChange(TopologyDescriptor newConfig) {
             ClusterDescriptor clusterInfo = newConfig.getStandbyClusters().get(clusterId);
             runtimeToRemoteCluster.get(clusterId).updateRouterClusterDescriptor(clusterInfo);
         }
+        updateRuntimeConfigId(newConfig);
```

Comment:
seems that in some cases the topology config id does not necessarily change, so you could avoid this change of updates if it remains the same.